torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1454,6 +1551,18 @@ void ONNXShapeTypeInference(
 
   SpecialPostProcess(n);
   ProcessConstantValueMap(n, opset_version);
+  if (n->kind() != prim::ListConstruct) {
```

Comment:
can these similar snippets merged into `SpecialPostProcess`, once `!IsSupportedNode(n)` is not early returning? My other PR will avoid early return anyway.

Reply:
Seems not. Updating reliable map should be the last step of shape inference. 
When we call `ConstantValueMap::SetShape` or `SetRank`, useInferredTypeMap is updated. This happens in the function `ProcessConstantValueMap`.
`SpecialPostProcess` is before `ProcessConstantValueMap`, we should not update reliable map there.